test/parallel/test-buffer.js

```diff
@@ -730,6 +730,14 @@ for (let i = 0; i < 256; i++) {
   assert.equal(hexb2[i], hexb[i]);
 }
 
+console.log('Create buffer from odd-length hex string should fail');
+assert.throws(function() {
+  new Buffer('8', 'hex');
+}, function(err) {
+  return err instanceof TypeError &&
+      err.message === 'The hex string must have an even number of characters';
```

Comment:
minor nit: `/^The hex string must have an even number of characters/.test(err.message)`